### PR TITLE
chore(docs): add repo related documents

### PR DESCRIPTION
**Pull request by mittachaitu**
This PR adds repo related documents and license checks

Signed-off-by: mittachaitu <sai.chaithanya@mayadata.io>

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -36,6 +36,14 @@ jobs:
     - name: Run Golint
       run: make golint
 
+  verify-license:
```
Comment (prateekpandey14) [inline .github/workflows/go.yml:39]: add as part of lint section ?

Comment (shubham14bajpai) [inline .github/workflows/go.yml:39]: Can we have the license check in the lint job itself? And the build job should run only if the lint passes.

Comment (prateekpandey14) [reply .github/workflows/go.yml:39]: change the order of action, 
- lint + license-check-->build

Comment (mittachaitu) [reply .github/workflows/go.yml]: Updated

Comment (mittachaitu) [reply .github/workflows/go.yml]: Done

**Review by shubham14bajpai — APPROVED** [review]
lgtm

**Review by prateekpandey14 — APPROVED** [review]
lgtm

**Pull request merged**